backend/fms_core/resources/extraction.py

```diff
@@ -210,6 +210,12 @@ def after_save_instance(self, instance, using_transactions, dry_run):
 
     def save_m2m(self, obj, data, using_transactions, dry_run):
         lineage = SampleLineage.objects.create(parent=self.extracted_from, child=obj, process_measurement=self.process_measurement)
+
+        # automatic project inheritance
+        for project in self.extracted_from.projects.all():
```

Comment:
self.extracted_from.projects.all() just raised a question in my mind ... does projects get the projects relation or the projects property of sample ? I think we should rename the relation (and the related_name) to make sure there is no ambiguity ... Maybe rename the manytomany relation on project to "samples_linked" and related_name to "projects_linked" that way we can be sure which one is used. In the case here, you point to the property and the all() would be redundant.

Reply:
I think it retrieves the projects property of sample, i.e. if that extracted sample is linked to PROJECT1 and PROJECT2 then projects.all() will be those 2 project objects. If the change is necessary do you want me to do it on this PR or create another one?

Reply:
well it is not directly linked to this PR so if you do not need to do it here...  If this uses the property the property already returns projects.all(). more .all() do not really hurt it is just unnecessary i think.

Reply:
Yup I wasnt too sure about this one too. Reimplementing a relationship with a property in Sample. First, in what case do we need to have that relationship reimplemented through a property instead of having the relationship directly?

Reply:
It gives a different tool : the property was made to provide a simplified queryset to the records, the relation provides an object list if i recall correctly ... This is not that useful but so are most properties ... I just wanted to make sure there was no name collision between them.

Reply:
Yes at least have different names for them so that you dont get one when you expect the other one and vice versa